python/ray/tune/trainable/trainable.py

```diff
@@ -1153,14 +1153,14 @@ def _create_logger(
     def _open_logfiles(self, stdout_file, stderr_file):
         """Create loggers. Open stdout and stderr logfiles."""
         if stdout_file:
-            stdout_path = os.path.expanduser(os.path.join(self._logdir, stdout_file))
+            stdout_path = (Path(self._logdir) / stdout_file).expanduser().as_posix()
```

Comment:
I'm wondering if we can just add this `Path(...).expanduser().as_posix()` logic once at the user entrypoint, then just assume posix paths everywhere else in our own code, rather than need to patch it everywhere internally.

Ex:
```
RunConfig(storage_path)
Tuner.restore(...), Trainer.restore(...)
RAY_AIR_LOCAL_CACHE_DIR
ExperimentAnalysis(path)
```

In the case of `self._logdir`, it should already be expanded/as_posix.

Reply:
The problem comes up when `Path` is used again on windows, which will convert a posix path into a `WindowsPath`. If we only want to have `as_posix()` in one place, we can't use pathlib downstream.